### ISSUE TITLE
order and dependencies in TestSuit

### ISSUE THREAD

**Issue by aerydna** [robotology/robot-testing#68]
According to @traversaro it would be important to control the order of execution of both tests and fixtures.

While usual unit testing has to be order independent, in a robot testing scenario it would be useful to test small components at first (like encoder response, plugin response etc) and then larger behaviors like for instance navigation and grasping.

For fixtures, it is necessary where some fixtures (speaking of the yarp framework a yarp device, yarp application or the yarpmanager itself for example) need other utilities (like yarp server) in order to start correctly.

As far as I know, this functionality is not available in RTF. The way to implement it is the following:
In my opinion, the simplest and most intuitive solution from both the user and the development side is to maintain the order of the xml tags (top-bottom). This can be done by simply change the tests and fixtures data storage for from std::set to std vector. While the loss in terms of performance is negligible in my opinion an inconsistency with the xml parsing policy of yarp (where only the depth order of the tags is relevant). However being RTF and YARP two completely separate frameworks this shouldn't be a problem.

### TLDR
There is no way to control the fixture or test execution order in RTF while in some cases this is a requirement. Let's discuss on how to implement it.

**Comment by traversaro** [comment]
cc @drdanz @randaz81 @pattacini

**Comment by traversaro** [comment]
Relevant discussion on stack overflow: https://stackoverflow.com/questions/1131495/in-xml-is-order-important . 
In general I find it makes sense to have different behavior between yarpmanager and rtf because `yarpmanager` launches external processes (whose order of execution is arbitrary by definition, depending on the OS scheduler) while RTF fixture and tests plugins are classes whose methods are launched in a sequential order by the `testrunner` process, and hence it make sense to define a sequence for them.

**Comment by aerydna** [comment]
fixed by #69
